### PR TITLE
API-22828-close-date-cst

### PR DESCRIPTION
**Pull request by stiehlrod**
## Summary
Adds close date to claims show

## Related issue(s)
[API-22828](https://vajira.max.gov/browse/API-22828)

## Testing done
Added new test

## What areas of the site does it impact?
ClaimsAPI: claimsController, claimsBlueprint

## Acceptance criteria

- [X]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X]  No error nor warning in the console.
- [ ]  Documentation has been updated (link to documentation)
- [X]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: modules/claims_api/spec/factories/auto_established_claims.rb [hunk 1]
```diff
@@ -175,4 +177,11 @@
     phase_type { 'Preparation for Decision' }
     phase_type_change_ind { '45' }
   end
+  factory :bnft_claim_lc_status_eight, class: OpenStruct do
+    max_est_claim_complete_dt { Faker::Time.backward(days: 5, period: :morning) }
+    min_est_claim_complete_dt { Faker::Time.backward(days: 7, period: :morning) }
+    phase_chngd_dt { Faker::Time.backward(days: 6, period: :morning) }
+    phase_type { 'Complete' }
+    phase_type_change_ind { '78' }
+  end
```
Comment (FonzMP) [inline modules/claims_api/spec/factories/auto_established_claims.rb:186]: I feel like we're getting excessive with these. let's loop back to this at a later date but I would recommend looking at adding traits in the near future to reduce duplication. 
https://github.com/thoughtbot/factory_bot/blob/main/GETTING_STARTED.md#traits

File: modules/claims_api/spec/requests/v2/veterans/claims_request_spec.rb [hunk 2]
```diff
@@ -600,6 +600,32 @@
...
+          it 'shows a closed date' do
+            with_okta_user(scopes) do |auth_header|
+              VCR.use_cassette('bgs/tracked_items/find_tracked_items') do
+                VCR.use_cassette('evss/documents/get_claim_documents') do
+                  expect_any_instance_of(BGS::EbenefitsBenefitClaimsStatus)
+                    .to receive(:find_benefit_claim_details_by_benefit_claim_id).and_return(bgs_claim)
+                  expect(ClaimsApi::AutoEstablishedClaim)
+                    .to receive(:get_by_id_and_icn).and_return(nil)
+
+                  get claim_by_id_path, headers: auth_header
```
Comment (mchristiansonVA) [inline modules/claims_api/spec/requests/v2/veterans/claims_request_spec.rb:615]: Just to confirm my understanding, the VCR cassettes return previously-recorded external API responses, which are then used for an actual call to the get claim by ID endpoint?

Comment (stiehlrod) [reply modules/claims_api/spec/requests/v2/veterans/claims_request_spec.rb:615]: yes, the recorded responses are used when running the tests. [Here](https://www.rubyguides.com/2018/12/ruby-vcr-gem/) is a reference.

**Review by FonzMP — APPROVED** [review]
:shipit:

**Review by acovrig — APPROVED** [review]

File: modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb [hunk 3]
```diff
@@ -153,6 +153,7 @@ def build_claim_structure(data:, lighthouse_id:, upstream_id:) # rubocop:disable
             claim_phase_dates: build_claim_phase_attributes(data, 'index'),
             claim_type_code: data[:bnft_claim_type_cd],
             claim_type: data[:claim_status_type],
+            close_date: data[:claim_complete_dt].present? ? format_bgs_date(data[:claim_complete_dt]) : nil,
```
[the remark this answers is not in the document]
Comment (stiehlrod) [reply modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb:156]: yes, I will make sure close_date uses the date_present method in the above mentioned PR. TY!

**Pull request merged**